api/user.go

```diff
@@ -19,11 +19,45 @@ type UserDatabase interface {
 	CreateUser(user *model.User) error
 }
 
+// UserChangeNotifier notifies listeners for user changes.
+type UserChangeNotifier struct {
+	userDeletedCallbacks []func(uid uint) error
```

Comment:
I doubt whether we need an `error` returned here.

An `error` is designed to propagate possibly recoverable errors to the upstream caller. In this case, the use of this callback is just to deliver a message that user info has changed, and the upstream caller can do nothing different on whether the callback returned a successful result, and the error is never handled in your implementation as well. I also cannot foresee a case that this error could be utilized in the future.

Reply:
https://github.com/gotify/server/blob/d89e6e5b81622b78637cd3218749382efc8802cb/api/user.go#L161-L165

So the user is actually created(on the database), but a 500 will be returned if an error is present. This is a little ambigous from my point.

My opinion is that, if you think we should stumble across the creation/deletion process even one of the callbacks failed, just use `log` to log an error on the server side but still returns an `200 OK`, which is my initial implementation. If we should abort, the record should not be written into the database at all.

Reply:
I think we have gone another way a little far, I did not see a problem in this so I think its OK to keep it for now.